### DISCUSSION TITLE
✅ iPad Pro terminal stops responding

### DISCUSSION THREAD

**Discussion by rkratochvil** [isontheline/pro.webssh.net#408] [General]
Hello,

When I’m using the WebSSH app, then switch over to another app such as chrome, then switch back to WebSSH the terminal doesn’t seem to respond. I cannot type anything in the terminal but I can hit the buttons line disconnect. Just wondering if this is normal? 

2021 IPad Pro 12.9”

**Comment by isontheline** [comment]
Hello @rkratochvil 👋

This behavior doesn't seem normal.

* On iOS 14?
* Does the keyboard dismiss icon works?

**Comment by rkratochvil** [comment]
Correct iOS 14, most up to date. 

Just realized I left out critical information. I’m using the apple magic keyboard. Just tested it now without using the keyboard and it works just fine. So this is something to do with the apple keyboard. 

I just tested the following: 
* Keyboard connected, fired up WebSSH, connected to my machine. 
* Keyboard still connected, switched over to chrome
* Keyboard still connected, switched back to WebSSH - No input works as stated before
* pulled my ipad off the keyboard. On screen keyboard works just fine
* Put ipad back on the magic keyboard, everything works. 

Very interesting. Anything else you want me to try? I really like this product and would like to help out in any way.

**Reply by isontheline** [reply]
These reproduce steps will help me a lot 🙏

I think it's also related to another issue I have with macOS release.

Thank you a lot for your support and feedback Ryan.

I will open an issue tonight from this discussion in order to track the progress.

**Reply by rkratochvil** [reply]
Sounds good. Let me know if there’s anything else you’d like me to test out. 🍻

**Reply by isontheline** [reply]
Closing https://github.com/isontheline/pro.webssh.net/issues/409 as iOS 15 seems to fix the issue

**Comment by d-rideout** [comment]
I have the same problem on an iPad with "Software Version 15.4" with an external keyboard.  (Does that mean iOS 15?)

I also like WebSSH for its relatively lightweight design philosophy, but this 'disconnect-upon-app-switching' makes it mostly unusable in practice.

**Reply by isontheline** [reply]
When switching back, does the terminal is empty? Or isn't cleared but just unresponsive?

**Reply by d-rideout** [reply]
It is not empty.  Just unresponsive.

**Reply by d-rideout** [reply]
Termius seems to have the same problem -- says that iOS only allows 20 seconds of suspension.

And offers to track location as a workaround.

**Reply by d-rideout** [reply]
At the moment WebSSH is responding after more than 20 seconds of suspension.  I will write again when it happens again.
